### DISCUSSION TITLE
Caddy as an alternative to Nginx

### DISCUSSION THREAD

**Discussion by herrdragon** [edgexfoundry/community#280] [General]
Hello,

Caddy it's a mature project and it makes it easier to write plugins since it is written in go.
Also, since EdgexFoundry started to use NATS, and NATS claim to be a replacement for Redis, did you/can you look into this implementation?

Thank you!

**Comment by cloudxxx8** [comment]
@herrdragon please provide more information and reference about Caddy, and we can see if there is any vulenteer to work on it.

**Reply by herrdragon** [reply]
Sure. [https://caddyserver.com/](https://caddyserver.com) It is very advanced with lots of good features. Most importantly, it is written in go and will fit better the edgex ecosystem. I believe it could replace nginx all together.
Anything about NATS?

Thank you

**Reply by cloudxxx8** [reply]
we support NATS as an optional message bus, but it won't be the only one option.
since the license change of redis, we may use MQTT as the default message bus in the future.

**Reply by cloudxxx8** [reply]
Thanks for the reference.
For the API Gateway replacement, we need the clear reasons, not just the programming language.  There is nothing bad in Nignx.
Cadday can be the alternitive option, not the replacement.  However, we need volumteer to write the UCR, ADR, and development.
https://docs.edgexfoundry.org/3.1/design/Process/
https://wiki.edgexfoundry.org/display/FA/Contributor%27s+Guide

**Comment by herrdragon** [comment]
Ok, I can look into UCR and ADR and see what I can contribute.

**Reply by herrdragon** [reply]
ADR submitted
